### PR TITLE
[Android.mk] add missing cpp files

### PR DESCRIPTION
**Pull request by simonlingoogle**

### CODE REVIEW

**Review by Irving-cl — APPROVED** [review]
LGTM

**Comment by bukepo** [comment]
add more build options to the Android build check?

**Comment by simonlingoogle** [comment]
> add more build options to the Android build check?

I was trying to build otbr-agent with 1.2 backbone router features and found these missing.
Maybe we can add BACKBONE_ROUTER option in Android.mk.

File: Android.mk [hunk 1]
```diff
@@ -110,6 +113,7 @@ LOCAL_SRC_FILES := \
     src/utils/event_emitter.cpp \
     src/utils/hex.cpp \
     src/utils/strcpy_utils.cpp \
+    src/utils/system_utils.cpp \
```
Comment (bukepo) [inline Android.mk:116]: To prevent such issues from happening again, android-check should be enhanced to turn 1.2 features on.

**Comment by simonlingoogle** [comment]
outdated

**Pull request closed without merging**